src/shared/ask-password-api.c

```diff
@@ -86,6 +87,34 @@ static int retrieve_key(key_serial_t serial, char ***ret) {
         return 0;
 }
 
+static int get_ask_password_directory_for_flags(AskPasswordFlags flags, char **ret) {
+        int r;
+
+        if (FLAGS_SET(flags, ASK_PASSWORD_USER))
+                return acquire_user_ask_password_directory(ret);
+
+        r = strdup_to(ret, "/run/systemd/ask-password/");
+        if (r < 0)
+                return r;
+
+        return 1; /* there's a suitable directory known to us */
```

Comment:
Not a matter, but `return strdup_to_full(...);`

Reply:
hmm, what do you mean?

Reply:
strdup_to_full() returns 1. That's tricky, so feel free to ignore.